firefox-ios/Client/Frontend/Summary/LiteLLMSummarizer.swift

```diff
@@ -0,0 +1,85 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+
+/// A wrapper around `LiteLLMClient` that provides summarized output (full or streamed)
+/// and normalizes underlying errors to `SummarizeError` type.
+public final class LiteLLMSummarizer: SummarizerProtocol {
+    private let client: LiteLLMClientProtocol
+    private let model: String
+    private let maxTokens: Int
+
+    public init(
+        client: LiteLLMClientProtocol,
+        model: String,
+        maxTokens: Int
+    ) {
+        self.client = client
+        self.model = model
+        self.maxTokens = maxTokens
+    }
+
+    /// Generates a full summary of the given text using the provided prompt.
+    /// - Parameters:
+    ///   - prompt: Instruction prompt to guide the summarization.
+    ///   - text: The text to be summarized.
+    /// - Returns: A summarized string
+    /// - Throws: `SummarizerError` if the request fails or if the response is invalid.
+    public func summarize(prompt: String, text: String) async throws -> String {
+        let options = LiteLLMChatOptions(model: model, maxTokens: maxTokens, stream: false)
+        // System message is used for the prompt, user message for the text.
+        let messages = [LiteLLMMessage(role: .system, content: prompt), LiteLLMMessage(role: .user, content: text)]
```

Comment:
nit
```suggestion
        let messages = [
             LiteLLMMessage(role: .system, content: prompt), 
             LiteLLMMessage(role: .user, content: text)
        ]
```

Reply:
I see its being used twice, so maybe we can DRY with 

```swift
var messages: [LiteLLMMessage] {
    return [
             LiteLLMMessage(role: .system, content: prompt), 
             LiteLLMMessage(role: .user, content: text)
        ]
} 
```

Reply:
Nice. Fixed in https://github.com/mozilla-mobile/firefox-ios/pull/28284/commits/142d11913082c9dbd4ee0afe64ae08abb809a8c9